File: tclpkg/gv/demo/modgraph.lua
#!/usr/bin/lua

-- display the kernel module dependencies

-- author: John Ellson <ellson@research.att.com>

require('gv')

modules = io.lines('/proc/modules') 

G = gv.digraph("G")
N = gv.protonode(G)
E = gv.protoedge(G)
gv.setv(G, 'rankdir', 'LR')
gv.setv(G, 'nodesep', '0.05')
gv.setv(N, 'shape', 'box')
gv.setv(N, 'width', '0')
gv.setv(N, 'height', '0')
gv.setv(N, 'margin', '.03')
gv.setv(N, 'fontsize', '8')
gv.setv(N, 'fontname', 'helvetica')
gv.setv(E, 'arrowsize', '.4')

for rec in modules do

   -- note - Lua's "string.gfind" in 5.1 became "string.gmatch" in 5.2

   for mod, usedbylist in string.gmatch(rec, "([_%w]+) %w+ %w+ ([-,_%w]+)") do
      n = gv.node(G, mod)
      for usedby in string.gmatch(usedbylist, "([-_%w]+)") do
         if (usedby ~= '-') and (usedby ~= '') then
            gv.edge(n, gv.node(G, usedby))
         end
      end
   end
end

gv.tred(G)

gv.layout(G, 'dot')
gv.render(G, 'gtk')

File: tclpkg/gv/demo/modgraph.php
#!/usr/bin/php
<?
#FIXME  - I don't know how to extend php's paths.  Needs:
#   ln -s /usr/lib64/graphviz/php/libgv_php.so /usr/lib64/php/modules/gv.so
#   ln -s /usr/lib64/graphviz/php/php/gv.php /usr/share/php/gv.php

include("gv.php");

# display the kernel module dependencies

# author: John Ellson <ellson@research.att.com>

$G = digraph("G");
$N = protonode($G);
$E = protoedge($G);

setv($G, "rankdir", "LR");
setv($G, "nodesep", "0.05");
setv($N, "shape", "box");
setv($N, "width", "0");
setv($N, "height", "0");
setv($N, "margin", ".03");
setv($N, "fontsize", "8");
setv($N, "fontname", "helvetica");
setv($E, "arrowsize", ".4");

$f = fopen("/proc/modules", "r");
while ( ! feof($f)) {
	$rec = fgets($f);

#    for mod, usedbylist in string.gfind(rec, "([_%w]+) %w+ %w+ ([-,_%w]+)") do
#       n = gv.node(G, mod)
#       for usedby in string.gfind(usedbylist, "([-_%w]+)") do
#          if (usedby ~= '-') and (usedby ~= '') then
#             gv.edge(n, gv.node(G, usedby))
#          end
#       end
#    end
#
}
fclose($f);

#layout($G, "dot");
#render($G, "png");

?>
